packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx

```diff
@@ -253,21 +272,25 @@ export const LivenessCameraModule = (
     (e: React.ChangeEvent<HTMLSelectElement>) => {
       const newDeviceId = e.target.value;
       const changeCamera = async () => {
+        setIsMetadataLoaded(false);
         const newStream = await navigator.mediaDevices.getUserMedia({
           video: {
             ...videoConstraints,
             deviceId: { exact: newDeviceId },
           },
           audio: false,
         });
-        send({
-          type: 'UPDATE_DEVICE_AND_STREAM',
-          data: { newDeviceId, newStream },
-        });
+        // Only update the stream and draw oval once metadata is loaded
+        if (isMetadataLoaded) {
+          send({
+            type: 'UPDATE_DEVICE_AND_STREAM',
+            data: { newDeviceId, newStream },
+          });
+        }
```

Comment:
How does run if `setIsMetaDataLoaded` is set to `false` in the same closure on line 275?

Reply:
This is updated now so it just sets `IsMetaDataLoaded` to false.